src/main/java/seedu/workbook/model/internship/DateTime.java

```diff
@@ -7,21 +7,22 @@
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeFormatterBuilder;
 import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
 
 /**
  * Represents a Date in WorkBook.
  * Guarantees: immutable; name is valid as declared in {@link #isValidDate(String)}
  */
 public class DateTime {
     public static final DateTime EMPTY_DATETIME = new DateTime("");
-    public static final String MESSAGE_CONSTRAINTS = "Date should be dd-mmm-yyyy hh:mm";
+    public static final String MESSAGE_CONSTRAINTS = "Date should be dd-mmm-yyyy hh:mm, and should be valid.";
 
-    private static final String datePattern = "d-MMM-yyyy HH:mm";
+    private static final String datePattern = "d-MMM-uuuu HH:mm";
     private static final DateTimeFormatterBuilder formatterBuilder = new DateTimeFormatterBuilder()
             .parseCaseInsensitive()
             .appendPattern(datePattern);
-    private static final DateTimeFormatter dateFormatter = formatterBuilder.toFormatter();
-
+    private static final DateTimeFormatter dateFormatter = formatterBuilder.toFormatter()
```

Comment:
so this will allow us to throw errors for non existent but correctly formatted dates?

Reply:
Yup, the part below `withResolverStyle(ResolverStyle.STRICT)` throws errors for invalid dates